### DISCUSSION TITLE
Field not being saved

### DISCUSSION THREAD

**Discussion by PeterKneale** [JasperFx/marten#2240] [General] [answered]
Unit tests demonstrate this odd behaviour and I'm not sure I understand why
- https://github.com/PeterKneale/MartenIssue
- When saving the below aggregate using the root service provider, members of _list *appear* to be preserved
- When saving the below aggregate using a scoped service provider, members of _list *appear* to be lost
- In both cases, the `_list` field is not written to the database

```cs
    public class Aggregate
    {
        private List<ValueObject> _list = new List<ValueObject>();

        .....
    }
    
    public class ValueObject
    {
       .....
    }
```
steps to reproduce 
1. clone https://github.com/PeterKneale/MartenIssue
2. docker-compose up
3. execute unit tests

In both cases, the `_list` field isnt written to the database
![image](https://user-images.githubusercontent.com/710594/170032052-888c168b-a8c0-486b-9d55-db8dedb961f0.png)

**Comment by mysticmind** [comment, marked as answer]
With ref. to discussion in Gitter, the following was suggested as a solution:
> A way to deal with that is to have a public getter/setter combo of `IEnumerable<T>` that delegates to the internal list. The whole "persistence ignorance" thing just doesn't work. Either you have to make compromises in your object structure for persistence, or use a separate domain model type for business logic than you use for persistence. 
> 
> use `public IEnumerable<ValueObject> List => _list;`, this is purely for persistence and reload of data to the private field.
